### PR TITLE
feat: request trial license from server

### PR DESCRIPTION
**Pull request by nnixaa**

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -3,14 +3,53 @@ GREEN='\033[0;32m'
...
-printf "${CYAN}Get UI Bakery license - ${GET_KEY_LINK}:\n${NC}"
-printf '\n'
-printf "Enter license key:\n"
-while read license; do
-  test "$license" != "" && break
-  printf "${RED}License key is required!${NC}\n"
-  printf "Enter license key:"
+printf "${CYAN}Starting license setup...\n${NC}"
+echo "Do you already have a UI Bakery license key?"
+echo "Y - enter your license key, N - start a free trial."
```
Comment (Tibing) [inline setup.sh:8]:
```suggestion
echo "Y - enter your license key, N - start a free trial. (Y)"
```
or something like that to indicated that Yes is a default.

File: setup.sh [hunk 2]
```diff
@@ -3,14 +3,53 @@ GREEN='\033[0;32m'
...
+  elif [[ "$have_license_key_y_n" == "N" ]] || [[ "$have_license_key_y_n" == "n" ]]; then
+    echo "Enter working email address to generate the free trial license of UI Bakery:"
+    while read email; do
+
+      email_regex="^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
+      if [[ $email =~ $email_regex ]] ; then
+        break
+      else
+        printf "${RED}Valid email is required to request a license${NC}\n"
+        printf "Enter email:"
```
Comment (Tibing) [inline setup.sh:22]:
```suggestion
        printf "Enter email:\n"
```

File: setup.sh [hunk 3]
```diff
@@ -3,14 +3,53 @@ GREEN='\033[0;32m'
...
+      fi
+    done
+
+    printf "EMAIL: ${email}\n\n"
+    CUSTOM_LICENSE_KEY="NO"
+    break
+  else
+   echo "Y - yes, I have have a license key; N - no, I want to start a free trial"
```
Comment (Tibing) [inline setup.sh:30]: looks like ought to be the same as line 8

**Review by Tibing — CHANGES_REQUESTED** [review]

File: setup.sh (hunk 1 above)
Comment (nnixaa) [reply setup.sh]: fixed

File: setup.sh (hunk 2 above)
Comment (nnixaa) [reply setup.sh]: fixed

File: setup.sh (hunk 3 above)
Comment (nnixaa) [reply setup.sh]: fixed

**Review by Tibing — APPROVED** [review]

**Pull request merged**